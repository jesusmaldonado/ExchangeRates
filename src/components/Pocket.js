import React from 'react'

export default function Pocket({currency, amount}){
  return (
      <tr>
        <td>{currency}</td>
        <td>{amount} </td>
      </tr>
  )
};
