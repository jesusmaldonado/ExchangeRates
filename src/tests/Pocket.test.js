import React from 'react';
import Pocket from '../components/pocket';
import renderer from 'react-test-renderer';

test('Pocket renders a currency and Amount', () => {
  const component = renderer.create(
    <Pocket currency="USD" amount="100"></Pocket>,
  );
  let tree = component.toJSON();
  const instance = component.root;
  const spans = instance.findAllByType('span');
  const [firstSpan, secondSpan] = spans;
  expect(firstSpan.props.className).toEqual(expect.stringContaining('flag-icon-us'));
  expect(secondSpan.children[0]).toEqual(expect.stringContaining("$100"));
  expect(tree).toMatchSnapshot();
})
