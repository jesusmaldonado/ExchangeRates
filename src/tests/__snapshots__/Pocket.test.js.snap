// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pocket renders a currency and Amount 1`] = `
<tr>
  <td
    className="text-center"
  >
    <span
      className="flag-icon flag-icon-us"
    />
  </td>
  <td
    className="text-center"
  >
    <span>
      $100
    </span>
  </td>
</tr>
`;
